python/pydantic_core/core_schema.py

```diff
@@ -3035,6 +3038,7 @@ def dataclass_field(
         name=name,
         schema=schema,
         kw_only=kw_only,
+        init=init,
         init_only=init_only,
```

Comment:
Just a side note - is this the equivalent of `init_var`? Interesting that we don't use the same naming, if so